Makefile

```diff
@@ -19,11 +19,42 @@ INSTALLINC = /usr/local/include
 # base CFLAGS
 CFLAGS += -Wall -Wextra -Wpedantic -std=c11 -I. -fPIC
 
+# MinGW compiler-specific flags
+ifneq "$(findstring mingw,$(CC))" ""
+	MINGW ?= 1
+endif
+ifeq ($(MINGW),1)
+  ifneq "$(findstring gcc,$(CC))" ""
+	CFLAGS += -Wno-unknown-pragmas
+  endif
+	MINGW_LIBS=-lshlwapi -lws2_32
+endif
+
+# Oracle compiler-specific flags
+ifneq "$(findstring suncc,$(CC))" ""
+	SUNPRO ?= 1
+endif
+ifeq ($(SUNPRO),1)
+	CFLAGS        += -fcommon
+	FORTIFY_FLAGS ?= -U_FORTIFY_SOURCE
```

Comment:
What does `?=` do? I never saw that before. Just `:=`

Reply:
?= means "set if unset in the environment".  ;)